### PR TITLE
Added Char extensions that covers all the static utility functions in System.Char.

### PR DESCRIPTION
**Pull request by SasLuca**
I was recently lamenting the fact that for many char operations I have to use static functions (eg: `char.ToLower(c)`) when methods seem like they would visually be so much more pleasing (eg: `c.ToLower()`).  I like the extensions this library provides and Char extensions seemed like a reasonable addition to this package given it already adds String extensions.

All unit tests still pass following my changes, I just added a file with the Char extensions. 
All extensions just call their equivalent static function from System.Char, so no additional unit tests are needed. 

I followed the style and conventions I saw in StringExtensions.cs and added a summary to every function. The comments are taken from System.Char with only slight modifications in places.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -690,3 +690,12 @@ var s = hex.FromHexString().ToUtf8String();
 var base32 = s.ToUtf8Bytes().Base32Encode();
 var s = base32.Base32Decode().ToUtf8String();
 ```
+
+#### Char:
+
+Wraps all static utility functions from System.Char:
```
Comment (rmandvikar) [inline README.md:696]: s/System.Char/`System.Char`/

File: src/rm.Extensions/CharExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+namespace rm.Extensions
+{
+	/// <summary>
+	/// Char extensions.
+	/// </summary>
+	public static class CharExtensions
+	{
+		/// <summary>
+		/// Determines whether a character is a digit.
+		/// </summary>
```
Comment (rmandvikar) [inline src/rm.Extensions/CharExtensions.cs:12]: i'd simply use below. similar for all extn methods.
```cs
		/// <inheritdoc cref="char.IsDigit(char)"/>
```

File: src/rm.Extensions/CharExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+			return char.IsLowSurrogate(c);
+		}
+		
+		/// <summary>
+		/// Return true for all characters below or equal U+00ff, which is ASCII + Latin-1 Supplement.
+		/// </summary>
+		public static bool IsLatin1(this char c)
+		{
+			return (uint)c <= '\x00ff';
+		}
```
Comment (rmandvikar) [inline src/rm.Extensions/CharExtensions.cs:112]: see https://referencesource.microsoft.com/#mscorlib/system/char.cs,82

File: src/rm.Extensions/CharExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+		
+		/// <summary>
+		/// Return true for all characters below or equal U+007f, which is ASCII.
+		/// </summary>
+		public static bool IsAscii(this char c)
+		{
+			return (uint)c <= '\x007f';
```
Comment (rmandvikar) [inline src/rm.Extensions/CharExtensions.cs:119]: see https://referencesource.microsoft.com/#mscorlib/system/char.cs,87

File: src/rm.Extensions/CharExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+		}
+
+		/// <summary>
+		/// Determines whether a character is lower-case.
+		/// </summary>
+		public static bool IsLower(this char c)
+		{
+			return char.IsLower(c);
+		}
+
```
Comment (rmandvikar) [inline src/rm.Extensions/CharExtensions.cs:137]: i see quite a few trailing whitespaces in this file. i use https://marketplace.visualstudio.com/items?itemName=MadsKristensen.TrailingWhitespaceVisualizer to remove those. please fix.

**Review by rmandvikar — COMMENTED** [review]
Typically, I avoid adding all extn methods for a type as it has a blast radius with every net framework bump. Also, issue like with `isAscii(char)` comes up which is added in net6.0+ (preprocessor directives can be used but become hard to manage). However, let's see how this change goes.

File: src/rm.Extensions/CharExtensions.cs (hunk 5 above)
Comment (SasLuca) [reply src/rm.Extensions/CharExtensions.cs:137]: Sure thing!

**Comment by rmandvikar** [comment]
I'll be addressing the pr comments shortly. I'm merging this pr, and will make the changes on top of it as I don't have perms to change the pr.

**Comment by rmandvikar** [comment]
I've made the changes that I mentioned in the pr comments, and the changes are on `dev` branch in commit 0234a43 ("char: Add char extensions", 2022-12-04) with you as author. The changes should be part of nuget `2.10.0`. As of now, I've cut out a prerelease version, `2.10.0-alpha1`, if you like to verify it. I'll cut out a release version sometime later. 

Thanks for the pr!

**Pull request merged**